java-client/session-examples/src/main/java/io/deephaven/client/examples/FetchObject.java

```diff
@@ -0,0 +1,74 @@
+package io.deephaven.client.examples;
+
+import io.deephaven.client.impl.ExportId;
+import io.deephaven.client.impl.FetchedObject;
+import io.deephaven.client.impl.HasTicketId;
+import io.deephaven.client.impl.Session;
+import picocli.CommandLine;
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.ExecutionException;
+
+@Command(name = "fetch-object", mixinStandardHelpOptions = true,
+        description = "Fetch object", version = "0.1.0")
+class FetchObject extends SingleSessionExampleBase {
+
+    @Option(names = {"--type"}, required = true, description = "The ticket type.")
+    String type;
+
+    @ArgGroup(exclusive = true, multiplicity = "1")
+    Ticket ticket;
+
+    @Option(names = {"-f", "--file"}, description = "The output file, otherwise goes to STDOUT.")
+    Path file;
+
+    @Option(names = {"-r", "--recursive"}, description = "If the program should recursively fetch.")
+    boolean recursive;
```

Comment:
Note that this breaks if there is a table anywhere in the mix, probably should special case the same types on the client as the server, at least "table" (since tablemap is almost certainly going to be an object like figure, and treetable too).
```
java.lang.IllegalArgumentException: No type registered for object class=io.deephaven.engine.table.impl.QueryTable, value=QueryTable

        at io.deephaven.server.object.ObjectServiceGrpcImpl.noTypeException(ObjectServiceGrpcImpl.java:111)
        at io.deephaven.server.object.ObjectServiceGrpcImpl.lambda$serialize$2(ObjectServiceGrpcImpl.java:78)
        at java.util.Optional.orElseThrow(Optional.java:408)
        at io.deephaven.server.object.ObjectServiceGrpcImpl.serialize(ObjectServiceGrpcImpl.java:78)
        at io.deephaven.server.object.ObjectServiceGrpcImpl.lambda$fetchObject$0(ObjectServiceGrpcImpl.java:67)
        at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:847)
```